compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -236,9 +238,15 @@ impl Inliner<'tcx> {
             return false;
         }
 
-        // Avoid inlining functions marked as no_sanitize if sanitizer is enabled,
-        // since instrumentation might be enabled and performed on the caller.
-        if self.tcx.sess.opts.debugging_opts.sanitizer.intersects(codegen_fn_attrs.no_sanitize) {
+        let self_features = &self.codegen_fn_attrs.target_features;
+        let callee_features = &codegen_fn_attrs.target_features;
+        if !callee_features.iter().all(|feature| self_features.contains(feature)) {
+            debug!("`callee has extra target features - not inlining");
+            return false;
+        }
+
+        if self.codegen_fn_attrs.no_sanitize != codegen_fn_attrs.no_sanitize {
```

Comment:
It seems like we could do inlining when a sanitizer is enabled in one function but disabled in another so long as that same sanitizer is disabled globally. Is that correct? If so, we could do something like the following, although handling this case optimally may not be worth the effort.

```suggestion
if self.tcx.sess.opts.debugging_opts.sanitizer.iter().all(|san| self.codegen_fn_attrs.no_sanitize.contains(san) == codegen_fn_attrs.no_sanitize.contains(san)) {
```

Reply:
I limited the comparison to enabled sanitizers.